### ISSUE TITLE
Feature Request: Renaming/Relocation of OneDrive Business shared folders

### ISSUE THREAD

**Issue by icchalmers** [abraunegg/onedrive#1251]
**Is your feature request related to a problem? Please describe.**
I use OneDrive Business Shared Folders. When a folder is shared with me, in Windows the shares are stored in a separate OneDrive folder and the folder names are contextualised by including the name of the person who created the share e.g. `{MyHomeDir}/OneDrive` and `{myhomedir}/OneDrive - {BusinessName}/{share user name} - {shared folder name}`.

The current OneDrive Linux client simply drops `{shared folder name}` straight into my personal OneDrive folder. This makes it hard to organise, lacks context for who shared which folder and has naming conflicts if two different users share a folder of the same name e.g. two students share a folder called "4th year project".

**Describe the solution you'd like**
It would be good if in the default behaviour was closer to the Windows client i.e. `{homedir}/OneDrive-{BusinessName}/{share user name} - {shared folder name}` was used as the storage location for each share.

Additionally it would be nice if I could redirect the shared folder to appear internal to my own OneDrive e.g. `{homedir}/OneDrive/SharedWithMe/{user}/{shared folders}`.

**Describe alternatives you've considered**
I considered trying to setup two instances of the OneDrive client and apply selective sync to each instance. While this would make shares external to my "personal" OneDrive folder, it would require me to create a new instance for every single user who shared a folder with me.

**Additional context**
If this functionality is already supported, it isn't documented in [BusinessSharedFolders.md](https://github.com/abraunegg/onedrive/blob/master/docs/BusinessSharedFolders.md).

**Comment by abraunegg** [comment]
@icchalmers 
As this would be a *major* shift in how OneDrive Business Shared Folder feature works, this will need very careful development & thought around implementation.

I am also surprised that this did not come up in the 9 odd months that the feature was a PR before inclusion into the main codebase - I guess not enough end user testing & feedback!

**Comment by abraunegg** [comment]
@icchalmers 
I am currently putting the finishing touches to onedrive v2.5.0-alpha-1 where this capability is being added|fixed

**Example: Online Folder State**
![image](https://github.com/abraunegg/onedrive/assets/4956234/f8198973-27c7-4219-ac15-9ccc0e6f29bf)

**Example: Windows Folder State**
![image](https://github.com/abraunegg/onedrive/assets/4956234/cb229370-e7b5-46c1-b80d-79b108071a5c)

**Example: Local Linux Folder State**
```
total 4
drwxrwxr-x.  7 alex alex  141 Sep 21 18:03  .
drwx------. 31 alex alex 4096 Sep 21 16:40  ..
drwx------.  2 alex alex    6 Sep 16 10:27  Attachments
drwx------.  4 alex alex   52 Sep 21 17:22 'Documents - testteamsite'
drwx------.  4 alex alex   43 Sep 21 17:02  Jenkins
drwx------.  6 alex alex  126 Sep 21 17:02 "Jenkins - testuser2 testuser2's files"
drwxrwxr-x.  2 alex alex   47 Jun  6 05:30  random_25k_files
```
**Sync Process:**
```
Reading configuration file: /home/alex/.config/onedrive/config
Invalid value for key in config file - using default value: application_id
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in: /home/alex/OneDrive
Fetching items from the OneDrive API for Drive ID: b!bO8V7s9SSk6r7mWHpIjURotN33W1W2tEv3OXV_oFIdQimEdOHR-1So7CqeT1MfHA ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins
Fetching items from the OneDrive API for Drive ID: b!fZgJhK-pU0eTQpylvmoYCkE4YgH_KRNDlxjRx9OWNqmV9Q_E_uWdRJKIB5L_ruPN ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins - testuser2 testuser2's files
Fetching items from the OneDrive API for Drive ID: b!BhWyqa7K_kqXqHtSIlsqjR5iJogxpWxDradnpVGTU2VxBOJh82Y6S4he4rdnGPBT ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Documents - testteamsite
Fetching items from the OneDrive API for Drive ID: b!knrQ8K6YIkqi-IrggfGWyi6Bu-HVzjxMpA7qvZX7Gy6-dlZIIb6fSo7k4cc0dlrV ..
Processing changes and items received from OneDrive ...
Performing a database consistency and integrity check on locally stored data ... 
Scanning local filesystem '~/OneDrive' for new data to upload ...
Perfoming final true up scan of online data from OneDrive
Fetching items from the OneDrive API for Drive ID: b!bO8V7s9SSk6r7mWHpIjURotN33W1W2tEv3OXV_oFIdQimEdOHR-1So7CqeT1MfHA ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins
Fetching items from the OneDrive API for Drive ID: b!fZgJhK-pU0eTQpylvmoYCkE4YgH_KRNDlxjRx9OWNqmV9Q_E_uWdRJKIB5L_ruPN ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins - testuser2 testuser2's files
Fetching items from the OneDrive API for Drive ID: b!BhWyqa7K_kqXqHtSIlsqjR5iJogxpWxDradnpVGTU2VxBOJh82Y6S4he4rdnGPBT ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Documents - testteamsite
Fetching items from the OneDrive API for Drive ID: b!knrQ8K6YIkqi-IrggfGWyi6Bu-HVzjxMpA7qvZX7Gy6-dlZIIb6fSo7k4cc0dlrV ..
Processing changes and items received from OneDrive ...

Sync with Microsoft OneDrive is complete
```

A rename of a folder online, triggers a rename locally:
```
Reading configuration file: /home/alex/.config/onedrive/config
Invalid value for key in config file - using default value: application_id
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in: /home/alex/OneDrive
Fetching items from the OneDrive API for Drive ID: b!bO8V7s9SSk6r7mWHpIjURotN33W1W2tEv3OXV_oFIdQimEdOHR-1So7CqeT1MfHA ..
Processing changes and items received from OneDrive ...
Moving ./Jenkins to ./Jenkins - RENAMED
Syncing this OneDrive Business Shared Folder: Jenkins - RENAMED
Fetching items from the OneDrive API for Drive ID: b!fZgJhK-pU0eTQpylvmoYCkE4YgH_KRNDlxjRx9OWNqmV9Q_E_uWdRJKIB5L_ruPN ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins - testuser2 testuser2's files
Fetching items from the OneDrive API for Drive ID: b!BhWyqa7K_kqXqHtSIlsqjR5iJogxpWxDradnpVGTU2VxBOJh82Y6S4he4rdnGPBT ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Documents - testteamsite
Fetching items from the OneDrive API for Drive ID: b!knrQ8K6YIkqi-IrggfGWyi6Bu-HVzjxMpA7qvZX7Gy6-dlZIIb6fSo7k4cc0dlrV ..
Processing changes and items received from OneDrive ...
Performing a database consistency and integrity check on locally stored data ... 
Scanning local filesystem '~/OneDrive' for new data to upload ...
Perfoming final true up scan of online data from OneDrive
Fetching items from the OneDrive API for Drive ID: b!bO8V7s9SSk6r7mWHpIjURotN33W1W2tEv3OXV_oFIdQimEdOHR-1So7CqeT1MfHA ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins - RENAMED
Fetching items from the OneDrive API for Drive ID: b!fZgJhK-pU0eTQpylvmoYCkE4YgH_KRNDlxjRx9OWNqmV9Q_E_uWdRJKIB5L_ruPN ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Jenkins - testuser2 testuser2's files
Fetching items from the OneDrive API for Drive ID: b!BhWyqa7K_kqXqHtSIlsqjR5iJogxpWxDradnpVGTU2VxBOJh82Y6S4he4rdnGPBT ..
Processing changes and items received from OneDrive ...
Syncing this OneDrive Business Shared Folder: Documents - testteamsite
Fetching items from the OneDrive API for Drive ID: b!knrQ8K6YIkqi-IrggfGWyi6Bu-HVzjxMpA7qvZX7Gy6-dlZIIb6fSo7k4cc0dlrV ..
Processing changes and items received from OneDrive ...

Sync with Microsoft OneDrive is complete
```

If you could review the above and provide feedback that would be greatly appreciated.

When onedrive v2.5.0-alpha-1 also lands, if you could *please* help test that would also be greatly appreciated.

**Comment by abraunegg** [comment]
@icchalmers 
OneDrive Client Version v2.5.0-alpha-1 is now available for testing - please review https://github.com/abraunegg/onedrive/discussions/2495 for details.

**Comment by icchalmers** [comment]
Thanks for implementing this @abraunegg. My only feedback is that it would be good if the extra step of needing to add shortcuts via the web interface could be avoided. I'm guessing that's easier said that done though, and as implemented will certainly get the job done.

I'm afraid I can't help with testing. I've since moved jobs and no longer have OneDrive Business, or even a box running Linux to test it on!  Thank you for your continued long-term support of OneDrive on Linux.

**Comment by abraunegg** [comment]
Marking as Fixed | Implemented as this is included in v2.5.0 alpha series codebase.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
